src/SlidingSyncManager.ts

```diff
@@ -323,4 +327,89 @@ export class SlidingSyncManager {
             firstTime = false;
         }
     }
+
+    /**
+     * Set up the Sliding Sync instance; configures the end point and starts spidering.
+     * The sliding sync endpoint is derived the following way:
+     *   1. The user-defined sliding sync proxy URL (legacy, for backwards compatibility)
+     *   2. The client `well-known` sliding sync proxy URL [declared at the unstable prefix](https://github.com/matrix-org/matrix-spec-proposals/blob/kegan/sync-v3/proposals/3575-sync.md#unstable-prefix)
+     *   3. The homeserver base url (for native server support)
+     */
+    public async setup(): Promise<SlidingSync | undefined> {
+        const client = MatrixClientPeg.safeGet();
+        const baseUrl = client.baseUrl;
+        const proxyUrl = SettingsStore.getValue("feature_sliding_sync_proxy_url");
+        const wellKnownProxyUrl = await this.getProxyFromWellKnown();
+
+        const slidingSyncEndpoint = proxyUrl || wellKnownProxyUrl || baseUrl;
+
+        this.configure(client, slidingSyncEndpoint);
+        logger.info("Sliding sync activated at", slidingSyncEndpoint);
+        this.startSpidering(100, 50); // 100 rooms at a time, 50ms apart
+
+        return this.slidingSync;
+    }
+
+    /**
+     * Get the sliding sync proxy URL from the client well known
+     * @return The proxy url
+     */
+    private async getProxyFromWellKnown(): Promise<string | undefined> {
+        const client = MatrixClientPeg.safeGet();
+        try {
+            const clientWellKnown = await AutoDiscovery.findClientConfig(client.baseUrl);
+        } catch (e) {
+            return undefined;
+        }
+        const proxyUrl = clientWellKnown?.["org.matrix.msc3575.proxy"]?.url;
```

Comment:
This is trying to access a value from inside the try scope, you'd need to do something like

```
let clientWellKnown: TYPE | undefined;
try {
    clientWellKnown = ...
} catch (e) {
    ...
}
const proxyUrl = ...